### PR TITLE
Add bound solution git monitor 

### LINKED ISSUE #3486: Add bound solution git monitor
### Description

The `GitEventsMonitor` class is low-level and just listens to the events from a specific repo.

We need a higher-level component that reacts as solutions are opened/closed to change the repo being tracked.

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #3486

### CODE REVIEW

File: src/Core/GitEventsMonitor.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ private void WatchGitEvents(string repoFolder, IFileSystemWatcherFactory fileSys
 
             fileSystemWatcher.Filter = HEADFile;
             
-            fileSystemWatcher.Changed += HeadFileChanged;
+            fileSystemWatcher.Renamed += HeadFileChanged;
```
Comment (duncanp-sonar) [inline src/Core/GitEventsMonitor.cs:68]: Why are we only looking at file-rename events?

Does this have anything to do with the main purpose of this PR, or is it a completely separate bug in the `GitEventsMonitor`? It looks more like the latter. If so, it would have been better create a separate bug for it and handle it in a separated PR and not mix unrelated changes.

Comment (ugras-ergun-sonarsource) [reply src/Core/GitEventsMonitor.cs:68]: Because when we change it in Git it does not throw a changed event. It creates a new file, deletes the old one and renames the new one to "HEAD"

File: src/ConnectedMode.UnitTests/BoundSolutionGitMonitorTests.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+            var gitWorkspaceService = CreateGitWorkSpaceService(repoPath);
+
+            var gitEventsMonitor = new Mock<IGitEvents>();
+
+            GitEventFactory gitEventFactory = (string path) =>
+            {
+                if (path != repoPath)
+                {
+                    throw new Exception();
```
Comment (duncanp-sonar) [inline src/ConnectedMode.UnitTests/BoundSolutionGitMonitorTests.cs:52]: It would be better to include a message in the error explaining that this is a test failure and why i.e. unexpected repo name supplied to the factory.

Alternatively, a simple approach would be just to store the supplied path and assert later that it matches the expected path.

File: src/ConnectedMode.UnitTests/BoundSolutionGitMonitorTests.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            var gitWorkspaceService = CreateGitWorkSpaceService(originalPath);
+
+            var originalEventsMonitor = new Mock<IGitEvents>();
+            var newEventsMonitor = new Mock<IGitEvents>();
+
+            GitEventFactory gitEventFactory = (string path) =>
+            {
+                if (path != originalPath && path != newPath)
+                {
+                    throw new Exception();
```
Comment (duncanp-sonar) [inline src/ConnectedMode.UnitTests/BoundSolutionGitMonitorTests.cs:83]: A descriptive error message would help debugging test failures.

**Review by duncanp-sonar — COMMENTED** [review]
A couple of comments/questions.

File: src/Core/GitEventsMonitor.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/Core/GitEventsMonitor.cs:68]: Do you think that is worth documenting in the code?
